tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/GraphQL/HotChocolate/HotChocolateCommon.cs

```diff
@@ -0,0 +1,170 @@
+                    span.ResourceName = $"{operationType} {operationName ?? "operation"}";
+                    span.SetTag(Trace.Tags.GraphQLOperationType, operationType);
+                }
+            }
+            catch (Exception ex)
+            {
+                Log.Error(ex, "Error updating HotChocolate scope.");
+            }
+        }
+
+        internal static void RecordExecutionErrorsIfPresent(Span span, string errorType, System.Collections.IEnumerable errors)
+        {
+            var executionErrors = GetList(errors);
+            var errorCount = executionErrors?.Count ?? 0;
+
+            if (errorCount > 0)
+            {
+                RecordExecutionErrors(span, errorType, errorCount, ConstructErrorMessage(executionErrors));
+            }
+        }
+
+        private static string ConstructErrorMessage(List<IError> executionErrors)
+        {
+            if (executionErrors == null)
+            {
+                return string.Empty;
+            }
+
+            var builder = Util.StringBuilderCache.Acquire(Util.StringBuilderCache.MaxBuilderSize);
+
+            try
+            {
+                const string tab = "    ";
+                builder.AppendLine("errors: [");
+
+                for (int i = 0; i < executionErrors.Count; i++)
+                {
+                    var executionError = executionErrors[i];
+
+                    builder.Append(tab).AppendLine("{");
+
+                    var message = executionError.Message;
+                    if (message != null)
+                    {
+                        builder.AppendLine($"{tab + tab}\"message\": \"{message.Replace("\r", "\\r").Replace("\n", "\\n")}\",");
+                    }
+
+                    builder.AppendLine($"{tab + tab}\"locations\": [");
+                    var locations = executionError.Locations;
+                    if (locations != null)
+                    {
+                        foreach (var location in locations)
+                        {
+                            if (location.TryDuckCast<ErrorLocationStruct>(out var locationProxy))
+                            {
+                                builder.AppendLine($"{tab + tab + tab}{{");
+                                builder.AppendLine($"{tab + tab + tab + tab}\"line\": {locationProxy.Line},");
+                                builder.AppendLine($"{tab + tab + tab + tab}\"column\": {locationProxy.Column}");
+                                builder.AppendLine($"{tab + tab + tab}}},");
+                            }
+                        }
+                    }
+
+                    builder.AppendLine($"{tab + tab}]");
+                    builder.AppendLine($"{tab}}},");
+                }
+
+                builder.AppendLine("]");
+            }
+            catch (Exception ex)
+            {
+                Log.Error(ex, "Error creating HotChocolate error message.");
+                return "errors: []";
```

Comment:
You should release the stringbuilder here as well
```suggestion
                Util.StringBuilderCache.Release(builder);
                return "errors: []";
```